File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.java
import java.io.*;
import java.util.Arrays;
import java.util.Deque;
import java.util.LinkedList;
import java.util.Queue;


public class Main{
    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String[] inputs = br.readLine().split(" ");
        int N = Integer.parseInt(inputs[0]);
        int M = Integer.parseInt(inputs[1]);

        Deque<Integer> deque = new LinkedList<>();

        for (int i=1; i<=N; i++) {
            deque.add(i);
        }

        bw.write("<");
        while (!deque.isEmpty()) {

            for (int i=0; i<M-1; i++) {
                deque.add(deque.pollFirst());
            }
            if (deque.size() == 1)
                bw.write(deque.pop()+"");
            else
                bw.write(deque.pop()+", ");
        }

        bw.write(">");
        bw.flush();


    }
}

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
import sys
from collections import deque

if __name__ == '__main__':

    N, K = map(int, sys.stdin.readline().split())

    q = deque([i for i in range(1, N+1)])
    ans = []
    while q:
        for _ in range(K):
            q.append(q.popleft())

        ans.append(str(q.pop()))

    print("<" + ", ".join(ans) + ">")